### PR TITLE
Add templates syntax highlighting 

### PR DESCRIPTION
**Pull request by variadico**
By default all .tmpl files will use the html/template syntax highlighting. However, you can apply highlighting to any file by using `:set syntax=gohtmltmpl`

### CODE REVIEW

**Comment by fatih** [comment]
@pi241a thanks for the PR. I'm going to look at it once I have some spare time. I also need to check and test it before I can merge it. Thanks for opening it :+1:

**Comment by pi241a** [comment]
Sweet. I'm not really experienced with Vim plugins or Vimscript, so definitely give it a code review.

File: ftdetect/gofiletype.vim [hunk 1]
```diff
@@ -21,3 +21,6 @@ endfunction
 au BufNewFile *.go setlocal filetype=go fileencoding=utf-8 fileformat=unix
 au BufRead *.go call s:gofiletype_pre()
 au BufReadPost *.go call s:gofiletype_post()
+
+au BufRead,BufNewFile *.tmpl set filetype=gohtmltmpl
+
```
Comment (cespare) [inline ftdetect/gofiletype.vim]: Extra newline.

File: syntax/gohtmltmpl.vim [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+if !exists("main_syntax")
+  let main_syntax = 'html'
+endif
+
+runtime! syntax/gotexttmpl.vim
+runtime! syntax/html.vim
+unlet b:current_syntax
+
+let b:current_syntax = "gohtmltmpl"
+
```
Comment (cespare) [inline syntax/gohtmltmpl.vim]: Extra newline

File: syntax/gotexttmpl.vim [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+syn region gotplAction start="{{" end="}}" contains=@gotplLiteral,gotplControl,gotplFunctions,gotplVariable,goTplIdentifier display
+syn region gotplAction start="\[\[" end="\]\]" contains=@gotplLiteral,gotplControl,gotplFunctions,gotplVariable display
+syn region goTplComment start="{{/\*" end="\*/}}" display
+syn region goTplComment start="\[\[/\*" end="\*/\]\]" display
+
+hi def link gotplAction PreProc
+hi def link goTplComment Comment
+
+let b:current_syntax = "gotexttmpl"
+
```
Comment (cespare) [inline syntax/gotexttmpl.vim]: Extra newline

**Comment by cespare** [comment]
You should also pull in indent.

**Comment by pi241a** [comment]
Okay. Removed the extra newlines.

Uh, not sure what you mean by "You should also pull in indent." Which files? Should I indent with spaces or tabs? How big should the indent be?

**Comment by fatih** [comment]
Hi @pi241a 

This is the setting I put in every file for the `vim-go` project:

```
" vim:ts=4:sw=4:et
```

**Comment by pi241a** [comment]
Got it. I added the modeline to the new files and reindented the code.

**Comment by fatih** [comment]
Thanks @variadico and @cespare. Awesome contribution :+1:

**Comment by cespare** [comment]
Sorry, I meant that you should pull in the indent file as well, not just syntax.

**Comment by cespare** [comment]
Should I open my own PR for that, or...?

**Comment by fatih** [comment]
@cespare would be awesome, do it please.

**Comment by cespare** [comment]
OK, will do in the morning.

**Comment by fatih** [comment]
Thanks @cespare

**Comment by cespare** [comment]
See #408

**Pull request merged**